src/planscape/forsys/raster_merger.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+from django.contrib.gis.gdal import GDALRaster
+
+
+# Merges multiple rasters into a single raster according to the scale and SRID
+# of the first raster that is merged.
+# Rasters are expected to be formatted offline. Thus, to keep the solution
+# simple, exceptions will be thrown if ...
+# ... a raster has skew != [0, 0]
+# ... a raster has no bands or more than 1 band
+# ... a raster's band has nodata_value != np.nan
+# ... an added raster has a different scale than the original raster
+# ... an added raster has a different srid than the original raster
+# ... an added raster has an overlapping pixel with the original raster
+class RasterMerger():
+    merged_raster: GDALRaster
+
+    def __init__(self, original_raster: GDALRaster = None) -> None:
+        self.merged_raster = None
+        if original_raster is not None:
+            self.add_raster(original_raster)
+
+    def add_raster(self, raster: GDALRaster) -> None:
+        self._ensure_raster_is_valid(raster)
+        if self.merged_raster is None:
+            self.merged_raster = raster
+        else:
+            self._ensure_raster_is_compatible_with_merged_raster(raster)
+            self._merge_raster(raster)
+
+    def _ensure_raster_is_valid(self, raster: GDALRaster) -> None:
+        if raster.skew.x != 0 and raster.skew.y != 0:
+            raise Exception(
+                "invalid raster skew, [%f, %f] (expected [0, 0])" %
+                (raster.skew.x, raster.skew.y))
+        if len(raster.bands) != 1:
+            raise Exception(
+                "invalid raster has %d bands (expected 1)" %
+                (len(raster.bands)))
+        if not np.isnan(raster.bands[0].nodata_value):
+            raise Exception(
+                "invalid raster has nodata value, %s (expected np.nan)" %
+                (raster.bands[0].nodata_value))
+
+    def _ensure_raster_is_compatible_with_merged_raster(
+            self, raster: GDALRaster) -> None:
+        scale = self.merged_raster.scale
```

Comment:
Maybe add a defensive check before this line?

  if self.merged_raster is None:
    return

Reply:
sure - done.
I've added the check to all the protected methods that refer to self.merged_raster.